drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/editor/BaseGuidedDecisionTableEditorPresenter.java

```diff
@@ -318,10 +319,9 @@ public void removeDocument(final GuidedDecisionTableView.Presenter dtPresenter)
     }
 
     void openOtherDecisionTable() {
+        decisionTableSelectedEvent.fire(DecisionTableSelectedEvent.NONE);
```

Comment:
The problem was that if the table removed from the graph was **not** the _selected_ table then firing the ```DecisionTableSelectedEvent``` would not cause the View to refresh fully; as the current state held the same table reference. 

I originally changed the code [here](https://github.com/manstis/drools-wb/blob/ea1ef8fc5fd6d624c6e9b3730777d9ebffc420b5/drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/GuidedDecisionTableModellerViewImpl.java#L372) to only disable the buttons if there was no active decision table; however I then found there were other places where "removal" _cleared_ the View state with no corresponding "set" state (as the ```DecisionTableSelectedEvent``` was not triggering a complete refresh).

To ensure all clear/set state cases are covered I therefore chose to fire a ```DecisionTableSelectedEvent.NONE``` event first (clearing/resetting the View state) before firing the correct ```DecisionTableSelectedEvent```. This will also better protect against future regressions (where we may add "remove state" logic that assumes "set state" is always in response to the event).